src/internal/pachsql/db.go

```diff
@@ -58,6 +92,85 @@ func OpenURL(u URL, password string) (*DB, error) {
 	return res, errors.EnsureStack(err)
 }
 
+// ListTables returns an array of SchemaTable structs that represent the tables.
+func ListTables(ctx context.Context, db *DB) ([]SchemaTable, error) {
+	rows, err := db.QueryContext(ctx, listTablesQuery)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not list tables")
+	}
+	defer rows.Close()
+	var tables []SchemaTable
+	if err := sqlx.StructScan(rows, &tables); err != nil {
+		return nil, errors.Wrap(err, "could not scan tables")
+	}
+	return tables, nil
+}
+
+// ScannedRows queries db and calls MapScanRows on the response rows.
+func ScannedRows(ctx context.Context, db *DB, query string) ([]RowMap, error) {
+	rows, err := db.QueryContext(ctx, query)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to query database")
+	}
+	defer rows.Close()
+	scannedRows, err := MapScanRows(rows)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to scan rows")
+	}
+	return scannedRows, nil
+}
+
+// MapScanRows iterates through rows and returns an array of sqlx.MapScan results. This is useful
+// for when you want to capture the results, but don't know what columns the query can return
+// or when you don't want to create a struct to represent the columns.
+func MapScanRows(rows *sql.Rows) ([]RowMap, error) {
```

Comment:
Yeah, I think this is just sqlx.Select.

Reply:
Nice, I should have read this library more before I wrote these functions. That means I can get rid of some of these